Makefile

```diff
@@ -4,11 +4,11 @@ VERSION := $(shell git describe --always --dirty)
 
 # Ghostunnel binary
 ghostunnel: $(SOURCE_FILES)
-	go build -ldflags '-X main.version=${VERSION}' -o ghostunnel .
+	go build -ldflags '-extldflags "-static" -X main.version=${VERSION}' -o ghostunnel .
```

Comment:
It's not recommended to build libltdl statically afaik, I'd prefer if we kept that as-is.

Reply:
I'm building my static application like this to make then run everywhere and ensure there is no dependency needed (eg. in `scratch` or `distroless` containers).

See https://medium.com/@diogok/on-golang-static-binaries-cross-compiling-and-plugins-1aed33499671
or http://blog.wrouesnel.com/articles/Totally%20static%20Go%20builds/

Reply:
Yeah, I think in general that's fine, but in this case one of Ghostunnel's dependencies from C land is libltdl (via libtool) so we can load PKCS#11 modules which are SO files at runtime. Statically linking libltdl is not best practice.

From the docs: "Avoid using the -static, -static-libtool-libs, or -all-static switches when linking programs with libltdl. This will not work on all platforms, because the dlopening functions may not be available for static linking."

See: https://www.gnu.org/software/libtool/manual/html_node/Distributing-libltdl.html